go/pkg/client/client_context.go

```diff
@@ -18,6 +18,19 @@ const (
 	remoteHeadersKey = "build.bazel.remote.execution.v2.requestmetadata-bin"
 )
 
+// LogContextInfof(ctx, x, ...) is equivalent to log.V(x).Infof(...) except it
+// also logs context metadata, if available.
+func LogContextInfof(ctx context.Context, v log.Level, format string, args ...interface{}) {
```

Comment:
nit: Can you rather extend the Verbose type with a function that logs the context info?

log.V(2).ContextInfof(...)?

Reply:
nvm this won't work with a package level function, and an embedded struct in the client context package is not really worth it. LGTM.